src/storage/data_table.cpp

```diff
@@ -210,6 +210,55 @@ TupleSlot DataTable::Insert(const common::ManagedPointer<transaction::Transactio
   return result;
 }
 
+RawBlock *DataTable::CreateCompactedBlock() {
+  RawBlock *block = NewBlock();
+  // This function is ONLY called during the compaction process, set the status to FREEZING.
+  block->controller_.GetBlockState()->store(BlockState::FREEZING);
+  return block;
+}
+
+bool DataTable::InsertIntoFreezingBlock(common::ManagedPointer<transaction::TransactionContext> txn,
```

Comment:
Do blocks under FREEZING status all have valid arrow headers? If so, should you also update the num_records in the header? I might have missed something.

Reply:
We decided to move away from this design, marking as resolved.